manifests/bosh-manifest/deployments/040-collectd.yml

```diff
@@ -0,0 +1,56 @@
+---
+releases:
+- name: collectd
+  sha1: 5e5c196f9c95fe12eb17bba0f40fdaecaddd6424
+  url: https://github.com/alphagov/paas-collectd-boshrelease/releases/download/0.4/collectd-0.4.tgz
```

Comment:
> The initial reason why we used fork was that we could add custom tags and pin our deployment to these tags.
> Now, because bosh-init can only consume tarballs (you can't upload a release to bosh-init like you can to BOSH), I had to create a dev-release tarbal (using standard bosh-cli way bosh create release --with-tarball --force --version 0.4, more documentation about bosh releases: https://bosh.io/docs/create-release.html#dev-release-release). I have created the tarballs from the same code as is tagged. That is, 0.4 release is created from code that's tagged as 0.4.
> 
> It seemed to me that a good place to put the release tarball was in github release. This practice is also used in some CF repositories (e.g. diego). We have a spike to analyse our situation with releases completely, but we did not want to block this story on it (#115142265 - see comments in the story). In the future, we can easily move the release location to elsewhere and we can change the way to build the releases.

That's good information. Please put it in the commits 👍 

> I think it should be instead described in the repo why we have a fork of it (e.g. in readme). It seems a bit odd to have to search through commits in other repos hoping for explanation.

It could work either way. I think that putting it in the commit message that uses it will be easier to maintain though, because the reasons for use may change over time and we're unlikely to keep the README up-to-date. In this particular case, our previous requirement to fork for stability and our new requirement to use it with bosh-init has meant that we've had to do something different, so it's relevant to that commit.

I don't think it's hard to find. Chances are you'd want to see where the release was used before you changed it, so you could use `git log -p -G <repo>` to find occurrences in our single repo.

> steps to create and upload release are a standard thing, documented on BOSH side or when you print help of bosh-cli. If we have a special workflow (e.g. where to put the releases), they should be described in ADR or team manual or some other generally accessible place, instead of random commit messages.

Nothing is standard if it's not automated. I hope it will be in the future after that SPIKE, but in the meantime we have no history of how it go there. What is standard and obvious to you may not be to the next person that has to work on it.

An example of this was the Terraform provider that we had to compile and vendor in 6f1014eaddc4bd32c01c5b595f0e7563c9a1842a - the next person that worked on it asked how they were created, even though they were familiar with Go, but was able to find all the information they needed from the commit.